app/services/cfe/create_state_benefits_service.rb

```diff
@@ -0,0 +1,44 @@
+module CFE
+  class CreateStateBenefitsService < BaseService
+    def cfe_url_path
+      "/assessments/#{@submission.assessment_id}/state_benefit"
+    end
+
+    def request_body
+      {
+        "state_benefits": build_transactions
+      }.to_json
+    end
+
+    private
+
+    def process_response
+      @submission.state_benefits_created!
+    end
+
+    def build_transactions
+      result = []
+
+      raise CFE::SubmissionError, 'Benefit transactions un-coded' if bank_transactions.keys.any?(nil)
+      raise CFE::SubmissionError, 'Benefit transaction cannot be identified' if bank_transactions.keys.map { |n| n.match?(/Unknown code/) }.any?
```

Comment:
I think sending unknown to CFE should be OK, it just records them as an state benefit of unknown type, and includes them in income.  Will check CFE.

Reply:
That is the intention anyway, so if it doesn't do that, we'll have to change CFE.  But we definitely want to send all state benefits over to CFE, whether we recognise them or not.